lib/sample_uploader/utils/importer.py

```diff
@@ -193,11 +190,25 @@ def _get_existing_sample(name, kbase_sample_id):
                 'read': row.get('read'),
                 'admin': row.get('admin')
             })
-        except SampleContentError as e:
+        except SampleContentWarning as e:
             e.row = row_num
-            errors.append(e)
+            warnings.warn(e)
+
+    user_keys = set()
+    for s in samples:
+        for n in s['sample']['node_tree']:
+            ukeys = set(n['meta_user'].keys())
+            ckeys = set(n['meta_controlled'].keys())
+            user_keys |= ukeys - ckeys
```

Comment:
never seen `|=` used before, nice. could you wrap whatever gets executed first in with parenthesis `()` for clarity?